mpas_analysis/ocean/histogram.py

```diff
@@ -0,0 +1,594 @@
+            componentName=parentTask.componentName,
+            tags=parentTask.tags,
+            subtaskName=f'plot_{fullSuffix}_{regionName}_{season}')
+
+        self.run_after(mpasClimatologyTask)
+        self.regionGroup = regionGroup
+        self.regionName = regionName
+        self.sectionName = sectionName
+        self.controlConfig = controlConfig
+        self.mpasClimatologyTask = mpasClimatologyTask
+        self.mpasMasksSubtask = mpasMasksSubtask
+        self.obsMasksSubtask = obsMasksSubtask
+        self.obsDicts = obsDicts
+        self.variableList = variableList
+        self.weightList = weightList
+        self.season = season
+        self.filePrefix = fullSuffix
+
+    def setup_and_check(self):
+        """
+        Perform steps to set up the analysis and check for errors in the setup.
+
+        Raises
+        ------
+        IOError
+            If files are not present
+        """
+
+        # first, call setup_and_check from the base class (AnalysisTask),
+        # which will perform some common setup, including storing:
+        #   self.inDirectory, self.plotsDirectory, self.namelist, self.streams
+        #   self.calendar
+        super(PlotRegionHistogramSubtask, self).setup_and_check()
+
+        self.xmlFileNames = []
+        for var in self.variableList:
+            self.xmlFileNames.append(
+                f'{self.plotsDirectory}/{self.filePrefix}_{var}_'
+                f'{self.regionName}_{self.season}.xml')
+
+    def run_task(self):
+        """
+        Plots histograms of properties in an ocean region.
+        """
+
+        self.logger.info(f"\nPlotting {self.season} histograms for "
+                         f"{self.regionName}")
+
+        config = self.config
+        sectionName = self.sectionName
+
+        calendar = self.calendar
+
+        main_run_name = config.get('runs', 'mainRunName')
+
+        region_mask_filename = self.mpasMasksSubtask.maskFileName
+
+        ds_region_mask = xarray.open_dataset(region_mask_filename)
+
+        mask_region_names = decode_strings(ds_region_mask.regionNames)
+        region_index = mask_region_names.index(self.regionName)
+
+        ds_mask = ds_region_mask.isel(nRegions=region_index)
+        cell_mask = ds_mask.regionCellMasks == 1
+
+        if len(self.obsDicts) > 0:
+            obs_region_mask_filename = self.obsMasksSubtask.maskFileName
+            ds_obs_region_mask = xarray.open_dataset(obs_region_mask_filename)
+            mask_region_names = decode_strings(ds_region_mask.regionNames)
+            region_index = mask_region_names.index(self.regionName)
+
+            ds_obs_mask = ds_obs_region_mask.isel(nRegions=region_index)
+            obs_cell_mask = ds_obs_mask.regionMasks == 1
+
+        in_filename = get_unmasked_mpas_climatology_file_name(
+            config, self.season, self.componentName, op='avg')
+        ds = xarray.open_dataset(in_filename)
+        ds = ds.where(cell_mask, drop=True)
```

Comment:
I'm seeing memory issues here:
```
Plotting JFM histograms for Pacific_Basin
analysis task oceanHistogram_plot_histogram_Ocean_Basins_Pacific_Basin_JFM failed during run
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.xylar/anvil/mambaforge/envs/test_mpas_analysis_py3.9/lib/python3.9/site-packages/mpas_analysis/shared/analysis_task.py", line 322, in run
    self.run_task()
  File "/gpfs/fs1/home/ac.xylar/anvil/mambaforge/envs/test_mpas_analysis_py3.9/lib/python3.9/site-packages/mpas_analysis/ocean/histogram.py", line 436, in run_task
    ds = ds.where(cell_mask, drop=True)
  File "/gpfs/fs1/home/ac.xylar/anvil/mambaforge/envs/test_mpas_analysis_py3.9/lib/python3.9/site-packages/xarray/core/common.py", line 1096, in where
    return ops.where_method(self, cond, other)
  File "/gpfs/fs1/home/ac.xylar/anvil/mambaforge/envs/test_mpas_analysis_py3.9/lib/python3.9/site-packages/xarray/core/ops.py", line 177, in where_method
    return apply_ufunc(
  File "/gpfs/fs1/home/ac.xylar/anvil/mambaforge/envs/test_mpas_analysis_py3.9/lib/python3.9/site-packages/xarray/core/computation.py", line 1192, in apply_ufunc
    return apply_dataset_vfunc(
  File "/gpfs/fs1/home/ac.xylar/anvil/mambaforge/envs/test_mpas_analysis_py3.9/lib/python3.9/site-packages/xarray/core/computation.py", line 480, in apply_dataset_vfunc
    result_vars = apply_dict_of_variables_vfunc(
  File "/gpfs/fs1/home/ac.xylar/anvil/mambaforge/envs/test_mpas_analysis_py3.9/lib/python3.9/site-packages/xarray/core/computation.py", line 422, in apply_dict_of_variables_vfunc
    result_vars[name] = func(*variable_args)
  File "/gpfs/fs1/home/ac.xylar/anvil/mambaforge/envs/test_mpas_analysis_py3.9/lib/python3.9/site-packages/xarray/core/computation.py", line 771, in apply_variable_ufunc
    result_data = func(*input_data)
  File "/gpfs/fs1/home/ac.xylar/anvil/mambaforge/envs/test_mpas_analysis_py3.9/lib/python3.9/site-packages/xarray/core/duck_array_ops.py", line 276, in where_method
    return where(cond, data, other)
  File "/gpfs/fs1/home/ac.xylar/anvil/mambaforge/envs/test_mpas_analysis_py3.9/lib/python3.9/site-packages/xarray/core/duck_array_ops.py", line 270, in where
    return _where(condition, *as_shared_dtype([x, y]))
  File "<__array_function__ internals>", line 180, in where
numpy.core._exceptions._ArrayMemoryError: Unable to allocate 15.2 GiB for an array with shape (1, 22852, 60, 2977) and data type float32
```
Given that this is happening with a pretty low-res mesh, I think it's certain to happen at higher res, too.

The solution is probably to just pull out the variables we actually need before masking:
```suggestion
        ds = ds[[f'timeMonthly_avg_{var}' for var in self.variableList]]
        ds = ds.where(cell_mask, drop=True)
```

Reply:
I think the out-of-memory problem is coming because it's applying the mask to variables on edges after broadcasting them to cells as well as edges (nonsense but it doesn't know that).